appengine/flexible/AntiForgery/Program.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) 2019 Google LLC.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+// 
+// http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+using Google.Cloud.Diagnostics.AspNetCore;
+using Microsoft.AspNetCore;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace AntiForgery
+{
+    public class Program
+    {
+        public static void Main(string[] args)
+        {
+            CreateWebHostBuilder(args).Build().Run();
+        }
+
+        public static IWebHostBuilder CreateWebHostBuilder(string[] args) =>
+            WebHost.CreateDefaultBuilder(args)
+                .UseDiagnostics()
+                .UseStartup<Startup>();
+    }
+
+    internal static class ProgramExtensions
+    {
+        internal static IWebHostBuilder UseDiagnostics(

```

Comment:
You could use `Platform.Instance()` from GAX to use our platform detection here. (I don't like the method name we have, but that's a separate annoyance.) `GaePlatformDetails` provides the  project ID, service ID and version ID.

Reply:
What will Platform.Instance() return when the user is running the application locally with a service account key?

Reply:
It will return a `Platform` with a `Type` of `PlatformType.Unknown`. So you'd still need to detect the project ID locally in that case (e.g. from an environment variable, but not *necessarily* the same one as AppEngine sets). We could potentially add detection via the project ID of the service account, for cases where a service account *has* been set.

I want to improve the `UseGoogleDiagnostics` call in various ways - we should definitely talk about requirements at some point. (It's too easy to gather them piecemeal and end up missing big use cases.)

This isn't a blocker for this PR though.